hedera-mirror-rosetta/build/Dockerfile

```
@@ -4,10 +4,19 @@
 
 FROM ubuntu:20.04 as builder
 ARG DEBIAN_FRONTEND=noninteractive
-# GIT_REF can be a branch or a tag
+# GIT_REF can be a branch, a tag, or a 40-charater git commit hash
 ARG GIT_REF=main
+ARG REPO_URL=https://github.com/hashgraph/hedera-mirror-node
 RUN apt-get update && apt-get install -y gcc git openjdk-17-jdk-headless
-RUN git clone -b ${GIT_REF} --depth 1 https://github.com/hashgraph/hedera-mirror-node
+RUN /bin/bash -c 'if [[ ${GIT_REF} =~ ^[0-9a-f]{40}$ ]]; then \
+      mkdir hedera-mirror-node && cd hedera-mirror-node; \
+      git init; \
+      git remote add origin ${REPO_URL}; \
+      git fetch --depth 1 origin ${GIT_REF}; \
+      git reset --hard FETCH_HEAD; \
+    else \
+      git clone -b ${GIT_REF} --depth 1 ${REPO_URL}; \
+    fi'
```

How about this which should work for both branches and commits?

```suggestion
RUN git clone --depth 1 -n "${REPO_URL}" && cd hedera-mirror-node && git fetch --depth 1 origin "${GIT_REF}" && git checkout "${GIT_REF}"
```